lib/coarse_op_mma.cuh

```diff
@@ -0,0 +1,633 @@
+                                                                                                stream);
+          break;
+        case 1:
+          launch_compute_uv_kernel<from_coarse, 1, QUDA_FORWARDS, bM, bN, bK, block_y, block_z>(tp, arg, min_threads,
+                                                                                                stream);
+          break;
+        case 2:
+          launch_compute_uv_kernel<from_coarse, 2, QUDA_FORWARDS, bM, bN, bK, block_y, block_z>(tp, arg, min_threads,
+                                                                                                stream);
+          break;
+        case 3:
+          launch_compute_uv_kernel<from_coarse, 3, QUDA_FORWARDS, bM, bN, bK, block_y, block_z>(tp, arg, min_threads,
+                                                                                                stream);
+          break;
+        default: errorQuda("arg.dim(=%d) is NOT supported.", arg.dim);
+        }
+      }
+    }
+
+    template <bool from_coarse, bool query_max = false, class Arg>
+    typename std::enable_if<!from_coarse, int>::type
+    launch_compute_uv_kernel(TuneParam &tp, const Arg &arg, int min_threads, const cudaStream_t &stream)
+    {
+      errorQuda("MMA implementation is ONLY built for !from_coarse.");
+      return -1;
+    }
+
+    /**
+        The following functions have switch's that list computeUV and computeVUV MMA kernels instantiations.
+        if query_max = true, it will simply return how many instantiations there are; if query_max = false,
+        the MMA kernel is launched with the corresponding configuration.
+     */
+
+    template <bool from_coarse, bool query_max = false, class Arg>
+    typename std::enable_if<Arg::fineColor == 6 && Arg::coarseColor == 6 && Arg::fineSpin == 2 && Arg::coarseSpin == 2, int>::type
+    launch_compute_uv_kernel(TuneParam &tp, const Arg &arg, int min_threads, const cudaStream_t &stream)
+    {
+      if (query_max) return 1;
+      switch (tp.aux.x) {
+      // clang-format off
+      case 0: launch_compute_uv_kernel<from_coarse,  16,  16,   8,   4,   8>(tp, arg, min_threads, stream); break;
+      case 1: launch_compute_uv_kernel<from_coarse,  16,  16,   8,   8,   4>(tp, arg, min_threads, stream); break;
+      // clang-format on
+      default:
+        errorQuda("tp.aux.x(=%d) is NOT supported by (%d, %d, %d, %d).", tp.aux.x, Arg::fineSpin, Arg::coarseSpin,
+                  Arg::fineColor, Arg::coarseColor);
+      }
+      return -1;
+    }
+
+    template <bool from_coarse, bool query_max = false, class Arg>
+    typename std::enable_if<Arg::fineColor == 24 && Arg::coarseColor == 24 && Arg::fineSpin == 2 && Arg::coarseSpin == 2,
+                            int>::type
+    launch_compute_uv_kernel(TuneParam &tp, const Arg &arg, int min_threads, const cudaStream_t &stream)
+    {
+#if (__COMPUTE_CAPABILITY__ > 700)
```

Comment:
This should be `>= 750` else this will erroneously use the Ampere/Turing kernels on Xavier (which is sm_72)

Reply:
Does Xavier have the same instruction set as Volta?

Reply:
Yes